src/core/utils/HttpResponse/decorators.ts

```diff
@@ -40,10 +40,19 @@ export function decorateResponse(
   // Cookie forwarding is only relevant in the browser.
   if (typeof document !== 'undefined') {
     // Write the mocked response cookies to the document.
-    // Note that Fetch API Headers will concatenate multiple "Set-Cookie"
-    // headers into a single comma-separated string, just as it does
-    // with any other multi-value headers.
-    const responseCookies = init.headers.get('Set-Cookie')?.split(',') || []
+    // Extract the Set-Cookie from the response header entry
+    // so that it can be parsed correctly even if the Set-Cookie value contains commas.
+    // This is a temporary solution until the getSetCookie method of Headers
+    // is available in TypeScript 5.2 or later.
+    const responseCookies = Array.from(init.headers.entries()).reduce<string[]>(
```

Comment:
Since we build with 5.2+, and consumers shouldn't be getting typed output for this, can we just use `headers.getSetCookie()` or does that actually cause a consumer failure downstream?

consumer typescript shouldn't _need_ to know I think, but we shouldbe able to still use it as long as `getSetCookie` is defined on the Headers prototype in the target environments

Reply:
```
{
  H: Object [Headers] {
    append: [Function: append],
    delete: [Function: delete],
    get: [Function: get],
    has: [Function: has],
    set: [Function: set],
    keys: [Function: keys],
    values: [Function: values],
    entries: [Function: entries],
    forEach: [Function: forEach]
  }
}
```

looks like our jsdom version doesn't have `getSetCookie` implemented, but typescript is fine with it

Reply:
It seems that the latest version of jsdom's Header does not yet implement getSetCookie.
https://github.com/jsdom/jsdom/blob/24.0.0/lib/jsdom/living/fetch/Headers-impl.js

I too think it is better to use `headers-polyfill` to deal with this temporarily.
But it seems that `HeadersPolyfill.prototype.getSetCookie.call(init.headers)` would cause eslint to error.

```
msw/src/core/utils/HttpResponse/decorators.ts
  0:0  error  Parsing error: DeprecationError: 'originalKeywordKind' has been deprecated since v5.0.0 and can no longer be used. Use 'identifierToKeywordKind(identifier)' instead

✖ 1 problem (1 error, 0 warnings)
```

Reply:
weird, that looks like `@typescript-eslint/parser` is probably failing somewhere and maybe a version update fixes it, i'll try landing an update

Reply:
https://github.com/mswjs/msw/pull/2084

I think updating the ts-eslint dependencies like in that PR should fix this error, which isn't related to the dependency at all, but the linter breaking

Reply:
Thanks, @mattcosta7 
I have changed to use `headers-polyfill`.